### ISSUE TITLE
Parsing guide's anchor tags are broken

### ISSUE THREAD

**Issue by ynx0** [urbit/urbit.org#1710]
Well, some of them. 
Specifically, everything under the [**Parser builders**](https://developers.urbit.org/guides/additional/parsing#parser-builders) section.

I think it's because the titles are also literal links into other parts of the reference, so the `id` ends up being `""`.

This also has the funny effect of making it appear as if your current heading is _all_ of the subheadings under _Parser builders_.

![image](https://user-images.githubusercontent.com/9275716/184053785-f45cae25-a197-46c7-b574-a60775a3e584.png)

**Comment by matildepark** [comment]
The anchors can be set manually by adding `{% #id-name %}` after the header and I think that’s just what we do here. That or assess why starting with a symbol deletes all text I suppose.

**Comment by matildepark** [comment]
So basically what was happening was that our ID constructor (that goes through the AST and grabs the string, manipulates it and squishes it to be web-friendly and to match the previous Markdown engines) was unable to see the string for cases where headings had code blocks in them because they weren't strictly trees, they were trees that also had arrays where their first children had an object with a child in it with the string.

Basically I [made it more recursive and to look for arrays and not just child properties](https://github.com/urbit/foundation-design-system/commit/cd17e9e634918274213da42f6e2c7326d5ff21a8#diff-be1589c1242e340580a45b71faf254c3ec6c542a21f565399eb84b0933e24cc0) in foundation-design-system, which alleviates this.
